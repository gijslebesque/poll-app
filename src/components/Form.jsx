import React from "react";
import PropTypes from "prop-types";

const Form = ({ children, name, icons }) => {
	return (
		<form className="field">
			<label className="label">{name}</label>
			<div className={`control ${icons}`}>{children}</div>
		</form>
	);
};

Form.prototype = {
	children: PropTypes.element.isRequired,
	name: PropTypes.string.isRequired,
	icons: PropTypes.string
};

export default Form;
