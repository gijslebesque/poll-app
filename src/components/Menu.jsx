import React, { memo } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const Menu = memo(({ polls }) => {
	return (
		<aside className="menu">
			<ul className="menu-list">
				<p className="menu-label">Polls</p>
				{polls.map((poll, i) => {
					return (
						<li key={i}>
							<Link className="menu-label" to={`/poll/${i}`}>
								{`${i + 1}. ${poll.question} by ${poll.owner}`}
							</Link>
						</li>
					);
				})}
			</ul>
		</aside>
	);
});

Menu.propTypes = {
	polls: PropTypes.array.isRequired
};

Menu.defaultProps = {
	polls: []
};

export default Menu;
