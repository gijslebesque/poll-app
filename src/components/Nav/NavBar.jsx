import React, { memo } from "react";
import { Link } from "react-router-dom";

const NavBar = memo(({ toggleModal, userName, logout }) => {
	return (
		<nav>
			<div className="navbar-menu">
				<div className="navbar-start">
					<Link className="navbar-item" to="/">
						Home
					</Link>

					<Link className="navbar-item" to="/all-polls">
						All polls
					</Link>

					<Link className="navbar-item" to="create-poll">
						Create poll
					</Link>
				</div>

				<div class="navbar-end">
					<div className="navbar-item">
						<div className="buttons">
							{userName && (
								<p className="button is-light" onClick={logout}>
									Logout
								</p>
							)}
							{!userName && (
								<p className="button is-light" onClick={toggleModal}>
									Log in
								</p>
							)}
						</div>
					</div>
				</div>
			</div>
		</nav>
	);
});

export default NavBar;
