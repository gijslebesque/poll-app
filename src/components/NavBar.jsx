import React, { memo, useState } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const NavBar = memo(({ toggleModal, userName, logout }) => {
	const [isActive, toggleBurger] = useState("");
	return (
		<nav>
			<span
				onClick={() => toggleBurger(!isActive ? "is-active" : "")}
				role="button"
				className={`navbar-burger burger ${isActive}`}
				aria-label="menu"
				aria-expanded="false"
			>
				<span aria-hidden="true" />
				<span aria-hidden="true" />
				<span aria-hidden="true" />
			</span>

			<div className={`navbar-menu ${isActive}`}>
				<div className="navbar-start">
					<Link className="navbar-item" to="/">
						Home
					</Link>

					<Link className="navbar-item" to="/create-poll">
						Create poll
					</Link>
				</div>

				<div className="navbar-end">
					<div className="navbar-item">
						<div className="buttons">
							{userName && (
								<p className="button is-light" onClick={logout}>
									Logout
								</p>
							)}
							{!userName && (
								<p className="button is-light" onClick={toggleModal}>
									Log in
								</p>
							)}
						</div>
					</div>
				</div>
			</div>
		</nav>
	);
});

NavBar.propTypes = {
	toggleModal: PropTypes.func.isRequired,
	userName: PropTypes.string.isRequired,
	logout: PropTypes.func.isRequired
};

export default NavBar;
