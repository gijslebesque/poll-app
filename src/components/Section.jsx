import React from "react";
import PropTypes from "prop-types";

const Section = ({ title, subtitle, children, style }) => {
	return (
		<section className="section" style={style}>
			<div className="container">
				<h2 className="title">{title}</h2>
				<h3 className="subtitle">{subtitle}</h3>
				{children}
			</div>
		</section>
	);
};

Section.propTypes = {
	title: PropTypes.string.isRequired,
	subtitle: PropTypes.string,
	style: PropTypes.object,
	children: PropTypes.any
};

export default Section;
