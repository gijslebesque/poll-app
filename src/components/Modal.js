import React, { useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUser } from "@fortawesome/free-solid-svg-icons";
import Form from "./Form";

const Modal = ({ toggleModal, setUserName }) => {
	const [name, setName] = useState("");

	return (
		<div className={`modal is-active`}>
			<div className="modal-background" />
			<div className="modal-content">
				<Form label="Username" icons="has-icons-left">
					<input
						className="input is-success"
						type="text"
						name="userName"
						placeholder="Text input"
						value={name}
						onChange={e => setName(e.target.value)}
					/>
					<span className="icon is-small is-left">
						<FontAwesomeIcon icon={faUser} />
					</span>
				</Form>
				<button
					onClick={e => setUserName("userName", name)}
					className="button is-link"
				>
					Submit
				</button>
			</div>
			<button
				onClick={toggleModal}
				className="modal-close is-large"
				aria-label="close"
			/>
		</div>
	);
};

export default Modal;
