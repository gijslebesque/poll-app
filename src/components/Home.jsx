import React from "react";
import Section from "./Section";

const Home = () => {
	return (
		<Section title="Welcome">
			<p>
				Ullamco et fugiat tempor esse ex. Exercitation laboris ex dolore dolor
				cupidatat labore occaecat consectetur fugiat. Occaecat occaecat occaecat
				tempor aliquip velit magna Lorem officia labore. Velit consectetur minim
				qui amet irure reprehenderit quis elit anim. Lorem magna nulla fugiat
				consectetur deserunt incididunt est officia occaecat nulla adipisicing
				deserunt ullamco ut. Mollit minim officia non sint aliqua aliquip
				voluptate. Culpa excepteur dolore commodo anim magna pariatur sint
				officia exercitation est consectetur aliqua excepteur.
			</p>
		</Section>
	);
};

export default Home;
