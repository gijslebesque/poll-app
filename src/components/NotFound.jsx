import React from "react";
import Section from "./Section";
import history from "../history";
const NotFound = () => {
	return (
		<Section title="404 Not found" subtitle="Sorry there's nothing here">
			<button className="button is-primary" onClick={() => history.push("/")}>
				Back home
			</button>
		</Section>
	);
};

export default NotFound;
