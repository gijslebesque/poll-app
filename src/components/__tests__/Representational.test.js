import React from "react";
import Form from "../Form";
import Home from "../Home";
import Menu from "../Menu";
import Modal from "../Modal";
import NavBar from "../NavBar";
import NotFound from "../NotFound";
import Section from "../Section";

import { render, shallow } from "enzyme";

describe("Representational components", () => {
	it("Form should render correctly", () => {
		const wrapper = render(<Form />);
		expect(wrapper).toMatchSnapshot();
	});

	it("Home should render correctly", () => {
		const wrapper = render(<Home />);
		expect(wrapper).toMatchSnapshot();
	});

	it("Menu should render correctly", () => {
		const wrapper = render(<Menu />);
		expect(wrapper).toMatchSnapshot();
	});

	it("Modal should render correctly", () => {
		const wrapper = render(<Modal />);
		expect(wrapper).toMatchSnapshot();
	});

	it("NavBar should render correctly", () => {
		const baseProps = {
			toggleModal: jest.fn(),
			userName: "",
			logout: jest.fn()
		};
		const wrapper = shallow(<NavBar {...baseProps} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("NotFound should render correctly", () => {
		const wrapper = render(<NotFound />);
		expect(wrapper).toMatchSnapshot();
	});

	it("Section should render correctly", () => {
		const wrapper = render(<Section />);
		expect(wrapper).toMatchSnapshot();
	});
});
