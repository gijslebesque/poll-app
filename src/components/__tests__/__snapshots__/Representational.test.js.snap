// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Representational components Form should render correctly 1`] = `ShallowWrapper {}`;

exports[`Representational components Home should render correctly 1`] = `ShallowWrapper {}`;

exports[`Representational components Menu should render correctly 1`] = `ShallowWrapper {}`;

exports[`Representational components Modal should render correctly 1`] = `ShallowWrapper {}`;

exports[`Representational components NavBar should render correctly 1`] = `ShallowWrapper {}`;

exports[`Representational components NotFound should render correctly 1`] = `ShallowWrapper {}`;

exports[`Representational components Section should render correctly 1`] = `ShallowWrapper {}`;
